oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCMetadataQueryDAO.java

```diff
@@ -129,14 +129,16 @@ public List<ServiceInstance> listInstances(Duration duration,
             .collect(toList());
     }
 
-    protected SQLAndParameters buildSQLForListInstances(String serviceId, long minuteTimeBucket, String table) {
+    protected SQLAndParameters buildSQLForListInstances(String serviceId, long minuteTimeBucket, long endMinuteTimeBucket, String table) {
         final var  sql = new StringBuilder();
         final var parameters = new ArrayList<>(5);
         sql.append("select * from ").append(table).append(" where ")
             .append(JDBCTableInstaller.TABLE_COLUMN).append(" = ?");
         parameters.add(InstanceTraffic.INDEX_NAME);
         sql.append(" and ").append(InstanceTraffic.LAST_PING_TIME_BUCKET).append(" >= ?");
         parameters.add(minuteTimeBucket);
+        sql.append(" and ").append(InstanceTraffic.LAST_PING_TIME_BUCKET).append(" <= ?");
```

Comment:
Is this the expected condition? Last ping time before end time?
How about an instance is still alive after end time, but registered?

Reply:
OK，I have modified it